Examples/IntelliTect.TestTools.Selenate.Examples/ElementsInteractions.cs

```diff
@@ -30,4 +30,18 @@ public void GetAListOfElementsFromCollection()
             7,
             foundCount);
     }
+
+    [Fact]
+    public void GetAnEnumerableOfElementHandlers()
+    {
+        DriverHandler.NavigateToPage("https://the-internet.herokuapp.com/challenging_dom");
+        IEnumerable<ElementHandler> foundHandlers = _ChallengingDomPage.Headers.GetElementHandlers();
+
+        Assert.True(foundHandlers.Count() > 1);
+
+        foreach (ElementHandler handler in foundHandlers)
+        {
+            Assert.True(handler.WaitForDisplayed());
+        }
```

Comment:
nit: The `.Count()` and the foreach cause multiple enumerations. Looking at how GetElementHandlers is implemented you may also want to insert a `.Take()` call before the ToList() to limit the number of times the enumerations is iterated.

```suggestion
        List<ElementHandler> foundHandlers = _ChallengingDomPage.Headers.GetElementHandlers().ToList();

        Assert.True(foundHandlers.Count > 1);

        foreach (ElementHandler handler in foundHandlers)
        {
            Assert.True(handler.WaitForDisplayed());
        }
```